Hestia/app/src/main/java/com/rugged/application/hestia/PeripheralListFragment.java

```diff
@@ -1,93 +1,231 @@
+                                 boolean isLastChild, View convertView, ViewGroup parent) {
+            final String childText = (String) getChild(groupPosition, childPosition);
+
+            if (convertView == null) {
+                LayoutInflater infalInflater = (LayoutInflater) getActivity()
+                        .getSystemService(Context.LAYOUT_INFLATER_SERVICE);
+                //inflate depending on toggle or slider
+                convertView = infalInflater.inflate(R.layout.child_list_item, null);
+            }
+
+            TextView txtListChild = (TextView) convertView.findViewById(R.id.child_item_text);
+            txtListChild.setText(childText);
+
+            ImageView imageview = (ImageView) convertView.findViewById(R.id.imageview);
+
+            imageview.setOnClickListener(new View.OnClickListener() {
+                @Override
+                public void onClick(View view) {
+                    PopupMenu popup = new PopupMenu(getActivity(), view);
+                    popup.getMenuInflater().inflate(R.menu.popup,
+                            popup.getMenu());
+
+                    for (int i = 0; i < getChildDevice(groupPosition, childPosition)
+                            .getActivators().size(); i++) {
+                        if (getChildDevice(groupPosition, childPosition).getActivators().get(i)
+                                .getType().equals("SLIDER")) {
+                            popup.getMenu().findItem(R.id.slide).setEnabled(true);
+                            popup.getMenu().findItem(R.id.slide).setVisible(true);
+                            break;
+                        }
+                    }
+
+                    popup.show();
+                    popup.setOnMenuItemClickListener(new PopupMenu.OnMenuItemClickListener() {
+                        @Override
+                        public boolean onMenuItemClick(MenuItem item) {
+                            switch (item.getItemId()) {
+                                case R.id.settings :
+                                    //Settings
+                                    break;
+                                case R.id.delete :
+                                    //Remove list item
+                                    break;
+                                case R.id.slide :
+                                    //show notification
+                                    final SlideDialog dialog = new SlideDialog(getActivity());
+                                    dialog.show();
+                                    break;
+                                default:
+                                    break;
+                            }
+                            return true;
+                        }
+                    });
+                }
+            });
+            return convertView;
         }
-    }
 
-    private class PeripheralAdapter extends RecyclerView.Adapter<PeripheralHolder> {
```

Comment:
Private classes will be split into their own respective classes, where they will receive proper documentation. This monolithic class will be refactored into multiple different units.